unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -4547,69 +4538,95 @@ protected SimObjPhysics[] GetAllVisibleSimObjPhysics(
+            if (filterObjectIds != null) {
+                foreach (string objectId in filterObjectIds) {
+                    if (!physicsSceneManager.ObjectIdToSimObjPhysics.ContainsKey(objectId)) {
+                        throw new ArgumentException(
+                            $"Object with id {objectId} does not exist in scene."
+                        );
+                    }
                 }
-            } else {
-                visSchemeEnum = this.visibilityScheme;
+                filterSimObjs = filterObjectIds
+                    .Select(objectId => physicsSceneManager.ObjectIdToSimObjPhysics[objectId])
+                    .ToList();
             }
-            return visSchemeEnum;
+
+            SimObjPhysics[] interactable;
+            SimObjPhysics[] visible = GetAllVisibleSimObjPhysicsDistance(
+                camera: camera,
+                maxDistance: maxDistance.GetValueOrDefault(this.maxVisibleDistance),
+                filterSimObjs: filterSimObjs,
+                interactable: out interactable
+            );
+
+            // #if UNITY_EDITOR
+            //             foreach (SimObjPhysics sop in visible)
+            //             {
+            //                 Debug.Log("Visible: " + sop.name);
+            //             }
+            // #endif
+
+            actionFinishedEmit(true, visible.Select(sop => sop.ObjectID).ToList());
         }
 
+        //old version of GetVisibleSimObjectsFromCamera
+        [ObsoleteAttribute(
+            message: "This action is deprecated. Call GetVisibleObjectsFromCamera instead.",
+            error: false
+        )]
         public void GetVisibleObjects(
             float? maxDistance = null,
             string visibilityScheme = null,
             int? thirdPartyCameraIndex = null,
             List<string> objectIds = null
         ) {
-            VisibilityScheme visSchemeEnum = getVisibilityScheme(visibilityScheme);
-
-            Camera camera;
+            Camera camera; //which camera are we checking visibility from?
             if (thirdPartyCameraIndex.HasValue) {
                 camera = agentManager.thirdPartyCameras[thirdPartyCameraIndex.Value];
-                if (visSchemeEnum != VisibilityScheme.Distance) {
-                    throw new System.NotImplementedException(
-                        $"Visibility scheme {visSchemeEnum} is not implemented for third party cameras. Must be 'distance'."
-                    );
-                }
-            } else {
+            }
+            //can also be used to query main camera
+            else {
                 camera = m_Camera;
             }
 
+            if (this.visibilityScheme != VisibilityScheme.Distance) {
+                throw new System.NotImplementedException(
+                    $"Visibility scheme {this.visibilityScheme} is not compatible. Visibility scheme should be 'Distance'."
+                );
+            }
+
```

Comment:
This is a bug. Someone can pass in `visibilityScheme="Collider"` and no error would be thrown because we're referencing `this.visibilityScheme` and not the local `visibilityScheme` parameter.

My recommendation would be to completely remove the `visibilityScheme` argument to this function. This will cause peoples' code to break if they specified it before but that's ok.